cpp/arduino/Arduino.h

```diff
@@ -5,6 +5,15 @@ Mock Arduino.h library.
 Where possible, variable names from the Arduino library are used to avoid conflicts
 
 */
+
+// hardware mocks
+volatile long long __ARDUINO_CI_SFR_MOCK[1024];
```

Comment:
I believe that if you defined a variable in a header file then it gets defined every time that header file is included and you get multiple definitions. Instead, if you define it as `extern` in the header and then put it in a `.cpp` file, it will be created only once.

Reply:
Also, I think this should declare a `uin8_t*` array, rather than a `long long` array. If not, I think that each byte in the original "address space" (i.e. offsets passed to `_SFR_IOx` macros) end up in different `long long` entries in this array, so multi-byte values do not map properly anymore (i.e. `ADC` maps to either `ADCH` or `ADCL` (dunno which off-hand), rather than being a combination of both).

Also, the current code violates strict type aliasing (accessing a `longlong[]` through a pointer of a different type is not allowed), which means the compiler *might* mess up optimizations regarding this array. I was hoping that making it a `uint8_t` array would solve this, but it seems this only works the other way around (accessing an arbitrary type through a `uint8_t*`). So maybe this needs access through a union to make it explicit to the compiler what's happening...